ppafm/cli/conv_rho.py

```diff
@@ -21,63 +22,63 @@ def handleNegativeDensity( rho ):
     rho[rho<0] = 0
     rho *= ( Q/rho.sum() )
 
-# ======== Main
+def main():
```

Comment:
So the intend here is to be able call this as a function from another script? Or what is the purpose/advantage to encapsulate everything inside main function?

Reply:
I do not much about how exactly this works, I just followed the template indicated by @NikoOinonen and used e.g. for `ppafm-generate-elff`, but... if you look inside the script `ppafm-conv-rho`, which gets installed by `pip` based on `pyproject.toml`, this script actually loads `conv_rho.py` as ``from ppafm.cli.conv_rho import main`` and then calls it by ``sys.exit(main())``. I have no idea if it could be done in a different way.

Reply:
There is a feature in setuptools called [_entry point_](https://setuptools.pypa.io/en/latest/userguide/entry_point.html), which allows easily adding executable scripts to the package distribution. The idea is that you define a function somewhere in your package, and then you automatically generate an installable script that simply imports and calls that function.